.circleci/config.yml

```diff
@@ -241,6 +242,7 @@ jobs:
             git diff --exit-code
   test_types:
     <<: *defaults
+    resource_class: 'medium+'
```

Comment:
CPU is used at 100% so why not https://app.circleci.com/pipelines/github/mui/material-ui/66091/workflows/f69ad922-5291-4f3e-97f7-5bab4af19a82/jobs/356686/resources, as far as I have looked at the CircleCI pricing, the price is linear with the build capacity. Moving from 2 CPU to 3 CPU is 50% more expensive, but if it's also 50% faster, than same price, faster feedback loop.

Reply:
`yarn typescript` utilizes 100% CPU even on my 20-thread machine, so this has the potential of scaling well.

Reply:
Then we could look into a higher class in the future, if this ever become a bottleneck neck to get the full CI status faster. I would only be worried about the time it takes to reach 100%, to keep it relatively small compared to the whole task.

Reply:
A third option might be to reduce the RAM limit for each node.js process https://medium.com/geekculture/node-js-default-memory-settings-3c0fe8a9ba1, https://nodejs.org/api/cli.html#cli_node_options_options. If garbage collection works well, we would use more % of the RAM on the host.